numpy/core/src/umath/override.c

```diff
@@ -96,23 +108,72 @@ normalize___call___args(PyUFuncObject *ufunc, PyObject *args,
             Py_DECREF(obj);
         }
     }
-    return 0;
+    /* finally, ufuncs accept 'sig' or 'signature' normalize to 'signature' */
+    return normalize_signature_keyword(*normal_kwds);
 }
 
 static int
-normalize_reduce_accumulate_args(PyUFuncObject *ufunc, PyObject *args,
-                                 PyObject **normal_args, PyObject **normal_kwds)
+normalize_reduce_args(PyUFuncObject *ufunc, PyObject *args,
+                      PyObject **normal_args, PyObject **normal_kwds)
 {
     /*
      * ufunc.reduce(a[, axis, dtype, out, keepdims])
+     */
+    int nargs = PyTuple_GET_SIZE(args);
+    int i;
+    PyObject *obj;
+    static char *kwlist[] = {"array", "axis", "dtype", "out", "keepdims"};
+
+    if (nargs < 1 || nargs > 5) {
+        PyErr_Format(PyExc_TypeError,
+                     "ufunc.reduce() takes from 1 to 5 positional "
+                     "arguments but %d were given", nargs);
+        return -1;
+    }
+    *normal_args = PyTuple_GetSlice(args, 0, 1);
+    if (*normal_args == NULL) {
+        return -1;
+    }
+
+    for (i = 1; i < nargs; i++) {
+        if (PyDict_GetItemString(*normal_kwds, kwlist[i])) {
+            PyErr_Format(PyExc_TypeError,
+                         "argument given by name ('%s') and position (%d)",
+                         kwlist[i], i);
+            return -1;
+        }
+        obj = PyTuple_GET_ITEM(args, i);
+        if (obj != Py_None) {
+            if (i == 3) {
+                obj = PyTuple_GetSlice(args, 3, 4);
+            }
+            PyDict_SetItemString(*normal_kwds, kwlist[i], obj);
+            if (i == 3) {
+                Py_DECREF(obj);
+            }
+        }
+    }
```

Comment:
Well, that was why I had `reduce` and `accumulate` together... I agree that it could be done better with helper functions, but would suggest to leave this for a follow-up PR -- which then could also address the duplication between override and normal paths (see #8892). I wouldn't be surprised is that made whatever I do here rather irrelevant...